### DISCUSSION TITLE
Where In with MongoDB Nested Document

### DISCUSSION THREAD

**Discussion by andrewl94** [prisma/prisma#21371] [Q&A] [answered]
### Question

I'm trying to do a select where in content must be equal of one of the values of an array, like the example here:

https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#get-user-records-where-the-name-can-be-found-in-the-following-list-saqui-clementine-bob

```
const getUser = await prisma.user.findMany({
  where: {
    name: { in: ['Saqui', 'Clementine', 'Bob'] },
  },
})

```

But i'm getting the following error form TypeScript when trying with nested document:

> Type '{ type: { in: string[]; }; }' is not assignable to type '(Without<AttributesCompositeFilter, AttributesObjectEqualityInput> & AttributesObjectEqualityInput) | (Without<...> & AttributesCompositeFilter) | undefined'.
  Types of property 'type' are incompatible.
    Type '{ in: string[]; }' is not assignable to type 'string'.ts(2322)

### How to reproduce (optional)

```
var arrayTypes : string[] = input.materialTypes;
const data = await prisma.page.findMany({
    take: limit,
    skip,
    include,
    where: {
      type:  { 
        type: {
            in: arrayTypes,
         },
       },
    },
    orderBy
});
```

Prisma schema:

```
model Page {
  id              String     @id @default(auto()) @map("_id") @db.ObjectId
  title           String
  excerpt         String
  content         String
  slug            String     @unique
  type            Attributes
  tags            String[]
  lang            String
  thumbnailUrl    String?
  authorId        String
  moduleId        String     @db.ObjectId
  module          Module     @relation(fields: [moduleId], references: [id])
  createdAt       DateTime
  updatedAt       DateTime
  files           File[]
  publicationDate DateTime
  authors         Authors[]
  customFields    Json?
  isPublic        Boolean
  isPublished     Boolean
}
type Attributes {
  title String
  type  String
  meta  Json?
}


```

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by andrewl94** [comment, marked as answer]
I figured out not long after posting.
I was missing the relation filter: https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#is

For reference:
```
var arrayTypes : string[] = input.materialTypes
typeValue = {
  is:{
    type:{in:arrayTypes}
  }
};
```

**Reply by nurul3101** [reply]
Thanks for taking the time to share the solution 🚀

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
